edweiss-app/app/(app)/courses/[id]/deck/[deckId]/index.tsx

```diff
@@ -0,0 +1,404 @@
+		});
+
+		setLoading(false);
+	}
+
+	const error_selected = existedDeckName ? 'This name has already been used' : emptyField ? 'Please fill in the field' : undefined;
+
+	return (
+		<>
+			<RouteHeader
+				title={deck?.data.name}
+				right={
+					<>
+						{current_user_type == 'professor' && <FancyButton backgroundColor='green' outlined loading={loading} onPress={() => {
+							publishDeck();
+						}}>
+							Publish Deck
+						</FancyButton>}
+						{allowToEditDeck(current_user_type, deck.data.ownerID, ids_professor_map) && <TTouchableOpacity
+							testID='toggleButton'
+							onPress={() => {
+								setName(deck?.data.name as string);
+								setShowDropdown(true);
+								cancelCardSelection();
+							}}
+							activeOpacity={0.2}
+						>
+							<Icon name={'settings'} size={30} />
+						</TTouchableOpacity>}
+					</>
+				}
+			/>
+
+			<Modal visible={showDropdown} animationType='fade' onRequestClose={() => setShowDropdown(false)}>
+				<TView flex={1} p={20} backgroundColor='mantle'>
+					<TView flexDirection="row" justifyContent="space-between" alignItems="center" mb={'lg'}>
+						<TTouchableOpacity testID='closeButton' alignItems="flex-start" onPress={() => { setShowDropdown(false); }}>
+							<Icon name={'close'} size={iconSizes.lg} color="blue" mr={8} />
+						</TTouchableOpacity>
+
+						<TView justifyContent='center' alignItems='center'>
+							<TText bold size='lg'>Option</TText>
+						</TView>
+
+						<TTouchableOpacity testID='shareButton' onPress={() => {
+							setShowDropdown(false);
+							router.push({
+								pathname: `courses/${courseId}/deck/${deckId}/shareDeckCard` as any,
+								params: {
+									type: "Deck",
+									indices_of_cards_to_share: JSON.stringify([NaN]) // empty array of indices
+								}
+							});
+						}}>
+							<Icon name={'share-social'} size={iconSizes.lg} color="blue" mr={8} />
+						</TTouchableOpacity>
+
+					</TView>
+
+
+
+					<TView my='md' mt={2} borderColor='crust' radius='lg'>
+						<FancyTextInput
+							value={name}
+							onChangeText={n => {
+								setName(n)
+								setEmptyField(false)
+								setExistedDeckName(false)
+							}}
+							placeholder='Enter the new name of the deck'
+							icon='bulb-outline'
+							label='Deck Name'
+							error={error_selected}
+							multiline
+							numberOfLines={3}
+						/>
+					</TView>
+
+					<FancyButton
+						outlined
+						//backgroundColor='transparent'
+						backgroundColor='blue'
+						onPress={() => {
+							updateDeck(name);
+						}}
+						mt={'md'}
+						mb={'md'}
+						style={{}}
+						icon='checkmark'
+					>
+						Update Deck Name
+					</FancyButton>
+
+					<FancyButton
+						outlined
+						//backgroundColor='transparent'
+						backgroundColor='cherry'
+						onPress={() => {
+							deleteDeck();
+						}}
+						style={{}}
+						icon='trash'
+					>
+						Delete this deck entirely!
+					</FancyButton>
+				</TView>
+			</Modal>
```

Comment:
I agree. But since the scale is not very big and considering the time to put this into an external file, I prefer to leave the modal in the place where it is called. But if the module scales up, I will definitely move it to an external file for readability and maintainability. Thanks for this great suggestion.